packages/react-ui-components/src/MultiSelectBox/multiSelectBox.js

```diff
@@ -148,7 +148,8 @@ class MultiSelectBox extends PureComponent {
         SelectBox: PropTypes.any.isRequired,
         IconComponent: PropTypes.any.isRequired,
         IconButtonComponent: PropTypes.any.isRequired,
-        MultiSelectBox_ListPreviewSortable: PropTypes.any.isRequired
+        MultiSelectBox_ListPreviewSortable: PropTypes.any.isRequired,
+        setActiveContentCanvasSrc: PropTypes.func
```

Comment:
Setting the content canvas src shouldn't be a concern of the SelectBox as its super specific and not reusable in this generic component which is also used in different contexts.

I would prefer when the SelectBox has a callback prop for clicking the header. This should be easily adjustable from what I see in your other code.
And then the reference editor which knows about Neos can have the handler which will then set the content canvas.